packages/core/src/app/shipping/PayPalFastlaneShippingAddress.tsx

```diff
@@ -22,20 +50,110 @@ export const PayPalFastlaneShippingAddress: FC<PayPalFastlaneShippingAddressProp
         shippingAddress,
+        addresses,
+        isGuest,
+        handleFieldChange,
+        consignments
     } = props;
+    const paypalFastlaneShippingComponent = useRef<PayPalFastlaneAddressComponentRef>({});
+    const paypalCommerceFastlaneOptions = {
+        paypalcommercefastlane: {
+            onPayPalFastlaneAddressChange: (
+                showPayPalFastlaneAddressSelector: PayPalFastlaneAddressComponentRef['showAddressSelector'],
+            ) => {
+                paypalFastlaneShippingComponent.current.showAddressSelector =
+                    showPayPalFastlaneAddressSelector;
+            },
+        },
+    };
 
-    return (
-        <PayPalFastlaneShippingAddressForm
-            address={shippingAddress}
-            countries={countries}
-            deinitialize={deinitialize}
-            formFields={formFields}
-            initialize={initialize}
-            isLoading={isLoading}
-            methodId={methodId}
-            onAddressSelect={onAddressSelect}
-            onFieldChange={onFieldChange}
-            onUnhandledError={onUnhandledError}
-        />
+    const braintreeFastlaneOptions = {
+        braintreefastlane: {
+            onPayPalFastlaneAddressChange: (
+                showPayPalFastlaneAddressSelector: PayPalFastlaneAddressComponentRef['showAddressSelector'],
+            ) => {
+                paypalFastlaneShippingComponent.current.showAddressSelector =
+                    showPayPalFastlaneAddressSelector;
+            },
+        },
+    };
+
+    const initializationOptions: ShippingInitializeOptions = isPayPalCommerceFastlaneMethod(
+        methodId,
     )
+        ? paypalCommerceFastlaneOptions
+        : braintreeFastlaneOptions;
+
+    const initializeShippingStrategyOrThrow = async () => {
+        try {
+            await initialize({
+                methodId,
+                ...initializationOptions,
+            });
+        } catch (error) {
+            if (typeof onUnhandledError === 'function' && error instanceof Error) {
+                onUnhandledError(error);
+            }
+        }
+    };
+
+    const deinitializeShippingStrategyOrThrow = async () => {
+        try {
+            await deinitialize({ methodId });
+        } catch (error) {
+            if (typeof onUnhandledError === 'function' && error instanceof Error) {
+                onUnhandledError(error);
+            }
+        }
+    };
+
+    useEffect(() => {
+        void initializeShippingStrategyOrThrow();
+
+        return () => {
+            void deinitializeShippingStrategyOrThrow();
+        };
+    }, []);
+
+    const { isPayPalFastlaneEnabled, paypalFastlaneAddresses, shouldShowPayPalFastlaneShippingForm } = usePayPalFastlaneAddress();
+
+    const shippingAddresses = isPayPalFastlaneEnabled && isGuest
+        ? paypalFastlaneAddresses
+        : addresses;
+
+    return (
+        <LoadingOverlay hideContentWhenLoading isLoading={!hasSelectedShippingOptions(consignments)}>
```

Comment:
I don't think this is the state that we should follow in this case. I some cases shipping option may not appear in shipping step, because it requires shipping address. Therefore, current implementation leads to infinite loading or shipping form for shoppers who do not have Fastlane account.

Please double check this moment